### PR TITLE
#471 Nested classes are not discovered by source generator

### PR DESCRIPTION
**Pull request by ekspedycja**
In current implementation of the Source Generator nested classes are not discovered properly. This may be an issue in case of some of Vertical Slices approaches where feature is a class that consists of multiple subclasses. 

```CS
public static partial class GetAssignedTasks
{
    public record Result()
    {
        public string ResultBlahBlah => "BlahBlahBlah";
    }

    public record Query()
        : TeamRequest<Result>
    {
    }

    public class Validator : AbstractValidator<Query>
    {
        public Validator()
        {
            RuleFor(p => p.GameId).NotEmpty();
            RuleFor(p => p.TeamId).NotEmpty();
            RuleFor(p => p.TeamName).NotEmpty();
        }
    }
    public partial class Endpoint : BaseTeamEndpoint<Query, Result, Result>
    {
        public override void Configure()
        {
            Get("assignedTasks");
        }
    }
}
```

Its not an issue with Endpoint and its inheritance as when class is moved outside the top level class it is discovered properly.

The PR also simplifies the approach with discovering namespaces - IMO current approach is the way around.

### CODE REVIEW

**Comment by nefarius** [comment]
Could you enable allowing edits by maintainers please? Trying to commit a minor style fix, thx.

File: Src/Generator/EndpointsDiscoveryGenerator.cs [hunk 1]
```diff
@@ -9,69 +9,42 @@ namespace FastEndpoints.Generator;
...
-            "Namotion",
-            "StackExchange",
-            "Grpc",
-            "PresentationFramework",
-            "PresentationCore",
-            "WindowsBase"
-    };
-
-    public void Initialize(IncrementalGeneratorInitializationContext context)
+     public void Initialize(IncrementalGeneratorInitializationContext context)
```
Comment (nefarius) [inline Src/Generator/EndpointsDiscoveryGenerator.cs:12]: Could you eliminate the one redundant space at the start, thanks 😎

**Review by nefarius — APPROVED** [review]

**Comment by dj-nitehawk** [comment]
@nefarius 

is this PR ok to be merged other than the space thing?
i didn't look at it properly, but saw that the exclusion list was removed.
is that alright?

**Comment by nefarius** [comment]
The exclusion list got removed; the matching of derived classes is now based on searching for the interfaces including the `FastEndpoints` namespace, I still wanna test it in one of my projects but IMHO unless it is not desired to allow someone to create endpoints in a `Microsoft`, `NuGet` etc. namespace it should work alright.

**Comment by nefarius** [comment]
I have a project where I deliberately had to avoid "NuGet" in my namespaces, this will be perfect to see if I can now do that 😎

**Comment by dj-nitehawk** [comment]
@ekspedycja @nefarius 

did some testing with this today and it was only discovering `public` types.
most of my endpoint classes are `internal sealed`. committed a change that will transform internal types as well.
i'll merge this PR now as it seems to be performing good.

@nefarius if you find any issues in your project, do open up a new issue and we'll look in to it then.

thanks a lot @ekspedycja for this 🙏

**Comment by nefarius** [comment]
> @nefarius if you find any issues in your project, do open up a new issue and we'll look in to it then.

Works!

```cs
namespace FastEndpoints
{
    public static class DiscoveredTypes
    {
        public static readonly global::System.Type[] All = new global::System.Type[]
        {
            typeof(global::NuGetCachingProxy.Endpoints.CatchallEndpoint),
            typeof(global::NuGetCachingProxy.Endpoints.PackageDownloadEndpoint),
        };
    }
}
```

👏

**Comment by ekspedycja** [comment]
I'm glad I was able to help 🙂

One thing came to my mind but I had no chance to test it. Currently the list of available types includes FluetValidation.IValidator. I am not quite sure if this may be an issue if some externals packages implement their own Fluent Validators. In the other hand the generator is run on users code not on imported nugets so this should be fine - or maybe a good idea to test that.

**Pull request merged**